mne/io/hitachi/tests/test_hitachi.py

```diff
@@ -195,21 +196,46 @@ def test_hitachi_basic(preload, version, n_ch, n_times, lowpass, sex, date,
         contents = contents.replace(b'\n', end)
     with open(fname, 'wb') as fid:
         fid.write(CONTENTS[version])
-    raw = read_raw_hitachi(fname, preload=preload)
+    raw = read_raw_hitachi(fname, preload=preload, verbose=True)
     data = raw.get_data()
     assert data.shape == (n_ch, n_times)
     assert raw.info['sfreq'] == 10
     assert raw.info['lowpass'] == lowpass
     assert raw.info['subject_info']['sex'] == sex
     assert np.isfinite(raw.get_data()).all()
     assert raw.info['meas_date'] == dt.datetime(*date, tzinfo=dt.timezone.utc)
-    # distances
+    # bad distances
     distances = source_detector_distances(raw.info)
-    want = [0.] * ((n_ch - 4) // 2)
-    assert_allclose(distances[::2], want, atol=0.)
+    want = [0.] * (n_ch - 4)
+    assert_allclose(distances, want, atol=0.)
+    raw_od_bad = optical_density(raw)
+    with pytest.warns(RuntimeWarning, match='will be zero'):
+        beer_lambert_law(raw_od_bad)
+    # good distances
+    mon = make_standard_montage('standard_1020')
+    if version == '1.18':
+        need = sum(([f'S{ii}', f'D{ii}'] for ii in range(1, 9)), [])[:-1]
+        have = 'F3 FC3 C3 CP3 P3 F5 FC5 C5 CP5 P5 F7 FT7 T7 TP7 P7'.split()
+        dist_tol = 0.01
+        bl_min = 1e-6
+    else:
+        need = sum(([f'S{ii}', f'D{ii}'] for ii in range(1, 18)), [])[:-1]
+        have = mon.ch_names[:len(need)]  # wrong but good enough for tests
+        dist_tol = 0.15  # really big because the list above is wrong
+        bl_min = 1e-8  # really small for the same reason
+    assert len(need) == len(have)
+    mon.rename_channels(dict(zip(have, need)))
+    raw.set_montage(mon)
+    distances = source_detector_distances(raw.info)
+    want = [0.03] * (n_ch - 4)
+    assert_allclose(distances, want, atol=dist_tol)
     test_rank = 'less' if n_times < n_ch else True
     _test_raw_reader(read_raw_hitachi, fname=fname,
                      boundary_decimal=1, test_rank=test_rank)  # low fs
+
+    # TODO: eventually we should refactor these to be in
+    # mne/io/tests/test_raw.py and run them for all fNIRS readers
```

Comment:
Can you not just add the Hitachi files to the existing parametize setting in `mne/preprocessing/nirs/tests`? e.g.

https://github.com/mne-tools/mne-python/blob/720a95c7599d933526b5db7fae823e4349176a01/mne/preprocessing/nirs/tests/test_beer_lambert_law.py#L30-L33

https://github.com/mne-tools/mne-python/blob/720a95c7599d933526b5db7fae823e4349176a01/mne/preprocessing/nirs/tests/test_scalp_coupling_index.py#L29-L32

Reply:
Not really easily because of the existing parametrize I think. Let's leave this as as TODO for another PR...

Reply:
Yeah I meant in future PR, but just as a note so that the above tests are expanded rather than reimplemented in `mne/io/tests/test_raw.py`